### PR TITLE
test: Code Coverage: FormButtonValidation Directive 

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 642a82c</samp>

Enabled unit testing for `FormButtonValidationDirective` by uncommenting and updating the test code in `form-button-validation.directive.spec.ts`. The test code uses a test component, TestBed, and spies to mock and verify the directive's behavior.

### <samp>🤖 Generated by Copilot at 642a82c</samp>

> _`FormButtonValidation`_
> _Tested with spies and TestBed_
> _Winter of bugs ends_

### <samp>🤖 Generated by Copilot at 642a82c</samp>

*  Uncommented and updated the unit test code for the `FormButtonValidationDirective` ([link](https://github.com/fylein/fyle-mobile-app/pull/2449/files?diff=unified&w=0#diff-596d5c273a66546b8d2c0118caf76f2d1abede8456b867ee6e9f1e7b748d3683L1-R143))

## Clickup
https://app.clickup.com/t/85ztznx1f

## Code Coverage
`100% Statements 32/32`
`94.73% Branches 18/19`
`100% Functions 9/9`
`100% Lines 31/31`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/shared/directive/form-button-validation.directive.spec.ts [hunk 1]
```diff
@@ -1,9 +1,143 @@
...
+  let directive: FormButtonValidationDirective;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [TestFormValidationButtonComponent, FormButtonValidationDirective],
+    });
+
+    fixture = TestBed.createComponent(TestFormValidationButtonComponent);
+    component = fixture.componentInstance;
+    buttonElement = fixture.debugElement.query(By.css('button'));
```
Comment (OmkarJ13) [inline src/app/shared/directive/form-button-validation.directive.spec.ts:25]: Use utility method from dom helpers for this

File: src/app/shared/directive/form-button-validation.directive.spec.ts [hunk 2]
```diff
@@ -1,9 +1,143 @@
...
+  });
+
+  describe('addLoader():', () => {
+    it('should add secondary loader', () => {
+      directive.buttonType = 'secondary';
+
+      directive.addLoader();
+
+      expect(directive.selectedElement.classList.contains('disabled')).toBeTrue();
+      expect(directive.selectedElement.innerHTML).toEqual('Save <div class="secondary-loader"></div>');
```
Comment (OmkarJ13) [inline src/app/shared/directive/form-button-validation.directive.spec.ts:96]: Use `innerText` here from `HTMLElement`. That will only return the visible text

File: src/app/shared/directive/form-button-validation.directive.spec.ts [hunk 3]
```diff
@@ -1,9 +1,143 @@
...
+    });
+
+    it('should add primary loader in prefix position', () => {
+      directive.buttonType = 'primary';
+      directive.loaderPosition = LoaderPosition.prefix;
+
+      directive.addLoader();
+
+      expect(directive.selectedElement.classList.contains('disabled')).toBeTrue();
+      expect(directive.selectedElement.innerHTML).toEqual('<div class="primary-loader"></div>Save');
```
Comment (OmkarJ13) [inline src/app/shared/directive/form-button-validation.directive.spec.ts:106]: Same as above

File: src/app/shared/directive/form-button-validation.directive.spec.ts [hunk 4]
```diff
@@ -1,9 +1,143 @@
...
+  });
+
+  it('resetButton(): should reset button if the loaded is added', () => {
+    directive.loaderAdded = true;
+
+    directive.resetButton();
+
+    expect(directive.selectedElement.classList.contains('disabled')).toBeFalse();
+    expect(directive.selectedElement.disabled).toBeFalse();
+    expect(directive.selectedElement.innerHTML).toEqual('undefined');
```
Comment (OmkarJ13) [inline src/app/shared/directive/form-button-validation.directive.spec.ts:117]: Why is this undefined? Would passing `defaultText` fix this?

File: src/app/shared/directive/form-button-validation.directive.spec.ts [hunk 5]
```diff
@@ -1,9 +1,143 @@
...
+      directive.loaderPosition = LoaderPosition.prefix;
+
+      directive.addLoader();
+
+      expect(directive.selectedElement.classList.contains('disabled')).toBeTrue();
+      expect(directive.selectedElement.innerHTML).toEqual('<div class="primary-loader"></div>Save');
+    });
+  });
+
+  it('resetButton(): should reset button if the loaded is added', () => {
```
Comment (OmkarJ13) [inline src/app/shared/directive/form-button-validation.directive.spec.ts:110]: loaded -> loader

**Review by Dimple16 — APPROVED** [review]

**Review by OmkarJ13 — APPROVED** [review]

**Pull request merged**